### PR TITLE
Add support Rails 7.1

### PR DESCRIPTION
**Pull request by osyo-manga**
Add suport Rails 7.1.
Since Rails 7.1 `#_find_record` refers to a record with `find_by!(@primary_key => id)` .
But if `@primary_key` is `"id"`, it can't refer to the intended record, so we hack it to refer to the record based on `self.class.bitemporal_id_key`.
see: https://github.com/rails/rails/blob/v7.1.0/activerecord/lib/active_record/persistence.rb#L1152-#L1171

### CODE REVIEW

**Comment by osyo-manga** [comment]
rails_main failing issue will be addressed in another PR.

**Review by mkmn — APPROVED** [review]

File: lib/activerecord-bitemporal/bitemporal.rb [hunk 1]
```diff
@@ -383,6 +383,18 @@ def destroy(force_delete: false, operated_at: Time.current)
         false
       end
 
+      if Gem::Version.new("7.1.0") <= ActiveRecord.version
+        # MEMO: Since Rails 7.1 #_find_record refers to a record with find_by!(@primary_key => id)
+        #       But if @primary_key is "id", it can't refer to the intended record, so we hack it to refer to the record based on self.class.bitemporal_id_key
+        #       see: https://github.com/rails/rails/blob/v7.1.0/activerecord/lib/active_record/persistence.rb#L1152-#L1171
+        def _find_record(*)
+          tmp_primary_key, @primary_key = @primary_key, self.class.bitemporal_id_key
```
Comment (alpaca-tc) [inline lib/activerecord-bitemporal/bitemporal.rb:391]: ぱっと見このコードはスレッドセーフではないのではない気がしたんですが、大丈夫ですかね？
勘違いであればごめんなさい！

Comment (osyo-manga) [reply lib/activerecord-bitemporal/bitemporal.rb:391]: @alpaca-tc Thanks. Can you be a more specific about which part is not thread-safe?

Comment (alpaca-tc) [reply lib/activerecord-bitemporal/bitemporal.rb:391]: I mistakenly thought `@primary_key` was a class context variable; it looks like an instance context variable, so no problem.

**Review by lighty — APPROVED** [review]
LGTM!!

**Pull request merged**